pkg/products/monitoring/reconciler.go

```diff
@@ -639,7 +639,7 @@ func (r *Reconciler) reconcileAlertManagerConfigSecret(ctx context.Context, serv
 	// handle smtp credentials
 	smtpSecret := &corev1.Secret{}
 	if err := serverClient.Get(ctx, types.NamespacedName{Name: r.installation.Spec.SMTPSecret, Namespace: rhmiOperatorNs}, smtpSecret); err != nil {
-		logrus.Warnf("could not obtain smtp credentials secret: %v", err)
+		return integreatlyv1alpha1.PhaseFailed, err
```

Comment:
This change is required as if the secret is not found the monitoring installation continues but the config for alert manager is built incorrectly without error in integreatly-operator but the incorrect config causes the alert manager pod to error out.

Reply:
So would the monitoring stage still complete if there is no smtp secret? :thinking: Just wondering cause I think work was done to allow installation to continue even if the smtp secret is not found and this would block installation at the monitoring stage :thinking: 
* https://github.com/integr8ly/integreatly-operator/pull/1160
* https://github.com/integr8ly/integreatly-operator/pull/865

Reply:
Good point @KevFan, I haven't tested it without smtp secret being present.

@philbrookes @pmccarthy wdyt?

Reply:
@KevFan no it wouldn't complete. But it's not obvious that there is some problem with the installation and the alert manager pod errors without any feedback to the user/installer.

Reply:
This code change is removed. Thanks for spotting this @KevFan 👍